WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt

```diff
@@ -0,0 +1,363 @@
+            AppLog.e(T.MAIN, "Unable to disable shift mode", e)
+        }
+    }
+
+    private fun assignNavigationListeners(assign: Boolean) {
+        setOnNavigationItemSelectedListener(if (assign) this else null)
+        setOnNavigationItemReselectedListener(if (assign) this else null)
+    }
+
+    override fun onNavigationItemSelected(item: MenuItem): Boolean {
+        val position = getPositionForItemId(item.itemId)
+        return if (position == PAGE_NEW_POST) {
+            handlePostButtonClicked()
+            false
+        } else {
+            setCurrentPosition(position, false)
+            mListener!!.onPageChanged(position)
+            true
+        }
+    }
+
+    private fun handlePostButtonClicked() {
+        val postView = getItemView(PAGE_NEW_POST)
+
+        // animate the button icon before telling the listener the post button was clicked - this way
+        // the user sees the animation before the editor appears
+        AniUtils.startAnimation(postView, R.anim.navbar_button_scale, object : Animation.AnimationListener {
+            override fun onAnimationStart(animation: Animation) {
+                // noop
+            }
+
+            override fun onAnimationEnd(animation: Animation) {
+                mListener!!.onNewPostButtonClicked()
+            }
+
+            override fun onAnimationRepeat(animation: Animation) {
+                // noop
+            }
+        })
+    }
+
+    override fun onNavigationItemReselected(item: MenuItem) {
+        // scroll the active fragment's contents to the top when user re-taps the current item
+        val position = getPositionForItemId(item.itemId)
+        if (position != PAGE_NEW_POST) {
+            val fragment = mNavAdapter!!.getFragment(position)
+            if (fragment is OnScrollToTopListener) {
+                (fragment as OnScrollToTopListener).onScrollToTop()
+            }
+        }
+    }
+
+    private fun getPositionForItemId(@IdRes itemId: Int): Int {
+        return when (itemId) {
+            R.id.nav_sites -> PAGE_MY_SITE
+            R.id.nav_reader -> PAGE_READER
+            R.id.nav_write -> PAGE_NEW_POST
+            R.id.nav_me -> PAGE_ME
+            else -> PAGE_NOTIFS
+        }
+    }
+
+    @IdRes
+    private fun getItemIdForPosition(position: Int): Int {
+        return when (position) {
+            PAGE_MY_SITE -> R.id.nav_sites
+            PAGE_READER -> R.id.nav_reader
+            PAGE_NEW_POST -> R.id.nav_write
+            PAGE_ME -> R.id.nav_me
+            else -> R.id.nav_notifications
+        }
+    }
+
+    private fun setCurrentPosition(position: Int, ensureSelected: Boolean) {
+        // new post page can't be selected, only tapped
+        if (position == PAGE_NEW_POST) {
+            return
+        }
+
+        // remove the title and selected state from the previously selected item
+        if (mPrevPosition > -1) {
+            showTitleForPosition(mPrevPosition, false)
```

Comment:
maybe use named param for the boolean value to increase readibility? 
`showTitleForPosition(prevPosition, show = false)`